components/scanners/trufflehog/internal/transformer/transformer.go

```diff
@@ -298,12 +306,20 @@ func (t *trufflehogTransformer) mapDataSource(ctx context.Context, location Truf
 	if location.SourceMetadata.Data.Filesystem.File == "" {
 		return "", errors.Errorf("unsupported trufflehog findings with empty file entry detected, finding %#v", location)
 	}
+
+	fullPath := location.SourceMetadata.Data.Filesystem.File
+	relativePath, err := filepath.Rel(t.stripFilePathPrefix, fullPath)
+	if err != nil {
+		// If there's an error, fall back to the full path to avoid losing data
+		relativePath = fullPath
```

Comment:
if there is an error over here we should not proceed, it means that there is probably a misconfiguration